File: ruby/lib/yourimageshare.rb
require_relative "yourimageshare/version"
require_relative "yourimageshare/errors"
require_relative "yourimageshare/types"
require_relative "yourimageshare/client"

# The official Ruby client for the YourImageShare upload API
# (https://yourimageshare.com/about/api). It mirrors the existing Go, JS
# (npm), and Python (PyPI) SDKs - same method names, same result shapes,
# same error semantics - just idiomatic Ruby on top (raises APIError
# instead of Go's (result, error) return style).
#
# Zero gem dependencies - only the Ruby standard library.
module YourImageShare
end

File: elixir/lib/yourimageshare.ex
defmodule YourImageShare do
  @moduledoc """
  Official Elixir client for the [YourImageShare](https://yourimageshare.com)
  upload API. See `YourImageShare.Client` for the full API - this module is
  just an entry point.

      client = YourImageShare.Client.new("YOUR_API_KEY")
      {:ok, result} = YourImageShare.Client.upload(client, "photo.jpg")
      result.direct
      #=> "https://yourimageshare.com/ib/aB3xY9qRz1"

  Full HTTP reference: https://yourimageshare.com/about/api
  """
end
